libr/include/r_util/r_event.h

```diff
@@ -0,0 +1,33 @@
+/* radare - LGPL - Copyright 2018 - pancake */
+
+#ifndef R_EVENT_H
+#define R_EVENT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct r_event_t {
+	void *user;
+	RList *callbacks;
+} REvent;
+
+typedef enum {
+	R_EVENT_META_SET = 0,
+} REventType;
+
+typedef struct r_event_meta_t {
+	int type;
+	ut64 addr;
+	const char *string;
+} REventMeta;
```

Comment:
This is specific for each REventType, isn't it?

Reply:
yes. the void* could be a union

Reply:
the problem i see is that we shold define ALL the event structs and types in util, and we cant register new types at runtime or over versions. this makes the event system dependant on the version of r2, which is not really a problem because we are already breaking apis frequently to ensure sanity :D